src/discussions/in-context-topics/TopicsView.test.jsx

```diff
@@ -0,0 +1,227 @@
+    renderComponent();
+
+    nonCoursewareTopics.forEach(async topic => {
+      const noncoursewareTopic = await screen.findByText(topic.name);
+
+      await act(async () => fireEvent.click(noncoursewareTopic));
+      await waitFor(async () => {
+        expect(screen.queryByText(topic.name)).toBeInTheDocument();
+        expect(lastLocation.pathname.endsWith(`/topics/${topic.id}`)).toBeTruthy();
+      });
+    });
+  });
+
+  it('A non-courseware topic should be on the top of the list', async () => {
+    await setupMockResponse();
+    renderComponent();
+    const topicsList = await screen.getByRole('list');
+    const { firstChild } = topicsList;
+    expect(within(firstChild).queryByText(nonCoursewareTopics[0].name)).toBeInTheDocument();
+    expect(topicsList.children[1]).toBe(topicsList.querySelector('.divider'));
+  });
+
+  it('A non-Courseware topic should have 3 stats and should be hoverable', async () => {
+    await setupMockResponse();
+    renderComponent();
+    const topicsList = await screen.getByRole('list');
+    const { firstChild } = topicsList;
+    const statsList = await firstChild.getElementsByClassName('pgn__icon');
+
+    expect(statsList.length).toBe(3);
+    fireEvent.mouseOver(statsList[0]);
+    expect(screen.queryByText('1 Discussion')).toBeInTheDocument();
+  });
+
+  it('Section groups should be listed in the middle of the topics list.', async () => {
+    await setupMockResponse();
+    renderComponent();
+    const topicsList = await screen.getByRole('list');
+    const sectionGroups = await screen.getAllByTestId('section-group');
+
+    expect(topicsList.children[1]).toStrictEqual(topicsList.querySelector('.divider'));
+    expect(sectionGroups.length).toBe(2);
+    expect(topicsList.children[5]).toStrictEqual(topicsList.querySelector('.divider'));
+  });
+
+  it('A section group should have only a title and required subsections.', async () => {
+    await setupMockResponse();
+    renderComponent();
+    const sectionGroups = await screen.getAllByTestId('section-group');
+    coursewareTopics.forEach(async (topic, index) => {
+      const statsList = await sectionGroups[index].getElementsByClassName('pgn__icon');
+      const subsectionGroups = await within(sectionGroups[index]).getAllByTestId('subsection-group');
+
+      expect(within(sectionGroups[index]).queryByText(topic.displayName)).toBeInTheDocument();
+      expect(statsList).toHaveLength(0);
+      expect(subsectionGroups).toHaveLength(2);
+    });
+  });
+
+  it('The subsection should have a title name, be clickable, and not have the stats', async () => {
+    await setupMockResponse();
+    renderComponent();
+    const subSection = await container.querySelector(`[data-subsection-id=${coursewareTopics[0].children[0].id}]`);
+    const subSectionTitleContainer = await within(subSection).queryByText(coursewareTopics[0].children[0].displayName);
+    const statsList = await subSection.getElementsByClassName('pgn__icon');
```

Comment:
Do not repeat yourself. // Best practice. 

const subsectionObject = coursewareTopics[0].children[0]
const subSection = await container.querySelector(`[data-subsection-id=${subsectionObject.id}]`);
const subSectionTitle = await within(subSection).queryByText(subsectionObject.displayName);